src/bit_array.cr

```diff
@@ -165,6 +165,54 @@ struct BitArray
     end
   end
 
+  # :inherit:
+  def fill(value : Bool) : self
+    return self if size == 0
+
+    if size <= 32
+      @bits.value = value ? ~(UInt32::MAX << size) : 0_u32
+    elsif size <= 64
+      @bits.as(UInt64*).value = value ? ~(UInt64::MAX << size) : 0_u64
+    else
+      to_slice.fill(value ? 0xFF_u8 : 0x00_u8)
```

Comment:
How much *do* the other two size-specific branches matter? Couldn't we just drop them? Or maybe drop one of them?

Reply:
Benchmarks: https://gist.github.com/HertzDevil/074c8edd88cf0d0e5238a633cace6689